tamr_client/dataset/dataframe.py

```diff
@@ -22,7 +25,7 @@ class AmbiguousPrimaryKey(Exception):
 def upsert(
     session: Session,
     dataset: Dataset,
-    df: pd.DataFrame,
+    df: "pd.DataFrame",
```

Comment:
This looks like the type/class `pd.DataFrame` is being replaced by a string, since this would require `pandas` to be imported otherwise.  Is this being evaluated in some way when typechecking and building the docs?

Reply:
Yes, this uses [forward references](https://www.python.org/dev/peps/pep-0484/#forward-references), which both `mypy` and `sphinx_autodoc_typehint` know how to interpret as a type annotation.